src/operator/quantization/mkldnn/mkldnn_quantized_fully_connect.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2018 by Contributors
+ * \file mkldnn_quantized_fully_connected.cc
+ * \brief
+ */
+
+#if MXNET_USE_MKLDNN == 1
+#include "../../nn/mkldnn/mkldnn_fully_connected-inl.h"
+#include "../quantization_utils.h"
+
+namespace mxnet {
+namespace op {
+
+void MKLDNNQuantizedFullyConnectedForward(const nnvm::NodeAttrs &attrs,
+                                          const OpContext &ctx,
+                                          const std::vector<NDArray> &in_data,
+                                          const std::vector<OpReqType> &req,
+                                          const std::vector<NDArray> &out_data) {
+  TmpMemMgr::Get()->Init(ctx.requested[fullc::kTempSpace]);
+  FullyConnectedParam param = nnvm::get<FullyConnectedParam>(attrs.parsed);
+  const size_t num_inputs = param.no_bias ? 2 : 3;
+
+  CHECK_EQ(in_data.size(), static_cast<size_t>(num_inputs * 3));
+  CHECK_EQ(out_data.size(), 3U);
+
+  NDArray data = in_data[fullc::kData];
+  NDArray weight = in_data[fullc::kWeight];
+  const TShape &ishape = data.shape();
+
+  // TODO(ciyong) int8 input will be enabled when mkldnn supports this data type
+  CHECK(data.dtype() == mshadow::kUint8)
+    << "MKLDNNQuantizedFullyConnected Op only supports uint8 for now, but got "
+    << mxnet::op::type_string(data.dtype());
+
+  if (ishape.ndim() != 2) {
+    CHECK(param.flatten)
+      << "QuantizedFullyConnected Op only supports flatten=true when ishape.ndim()!=2 for now.";
+    data = data.MKLDNNDataReshape(Shape2(ishape[0], ishape.ProdShape(1, ishape.ndim())));
+  }
+
+  const float min_data = in_data[num_inputs].data().dptr<float>()[0];
+  const float max_data = in_data[num_inputs + 1].data().dptr<float>()[0];
+  const float min_weight = in_data[num_inputs + 2].data().dptr<float>()[0];
+  const float max_weight = in_data[num_inputs + 3].data().dptr<float>()[0];
```

Comment:
Use meaningful enums like kData, kWeight, kBias instead of 1, 2, 3, or add comments about how input vector is arranged.

Reply:
no problem :)